pyraphtory/src/graph_window.rs

```diff
@@ -73,21 +102,95 @@ impl WindowedVertex {
         self.vertex_w.out_degree()
     }
 
-    pub fn neighbours(&self) -> EdgeIterator {
-        EdgeIterator {
-            iter: Box::new(self.vertex_w.neighbours().map(|te| te.into())),
+    pub fn edges(&self) -> WindowedEdgeIterator {
+        WindowedEdgeIterator {
+            iter: Box::new(self.vertex_w.edges().map(|te| te.into())),
         }
     }
 
-    pub fn in_neighbours(&self) -> EdgeIterator {
-        EdgeIterator {
-            iter: Box::new(self.vertex_w.in_neighbours().map(|te| te.into())),
+    pub fn in_edges(&self) -> WindowedEdgeIterator {
+        WindowedEdgeIterator {
+            iter: Box::new(self.vertex_w.in_edges().map(|te| te.into())),
         }
     }
 
-    pub fn out_neighbours(&self) -> EdgeIterator {
-        EdgeIterator {
-            iter: Box::new(self.vertex_w.out_neighbours().map(|te| te.into())),
+    pub fn out_edges(&self) -> WindowedEdgeIterator {
+        WindowedEdgeIterator {
+            iter: Box::new(self.vertex_w.out_edges().map(|te| te.into())),
         }
     }
+
+    pub fn neighbours(&self) -> WindowedVertexIterator {
+        WindowedVertexIterator {
+            iter: Box::new(self.vertex_w.neighbours().map(|tv| tv.into())),
+        }
+    }
+
+    pub fn in_neighbours(&self) -> WindowedVertexIterator {
+        WindowedVertexIterator {
+            iter: Box::new(self.vertex_w.in_neighbours().map(|tv| tv.into())),
+        }
+    }
+
+    pub fn out_neighbours(&self) -> WindowedVertexIterator {
+        WindowedVertexIterator {
+            iter: Box::new(self.vertex_w.out_neighbours().map(|tv| tv.into())),
+        }
+    }
+
+    pub fn neighbours_ids(&self) -> VertexIdsIterator {
+        VertexIdsIterator {
+            iter: Box::new(self.vertex_w.neighbours_ids()),
+        }
+    }
+
+    pub fn in_neighbours_ids(&self) -> VertexIdsIterator {
+        VertexIdsIterator {
+            iter: Box::new(self.vertex_w.in_neighbours_ids()),
+        }
+    }
+
+    pub fn out_neighbours_ids(&self) -> VertexIdsIterator {
+        VertexIdsIterator {
+            iter: Box::new(self.vertex_w.out_neighbours_ids()),
+        }
+    }
+}
+
+#[pyclass]
+pub struct WindowedEdge {
+    pub edge_id: usize,
+    #[pyo3(get)]
+    pub src: u64,
+    #[pyo3(get)]
+    pub dst: u64,
+    #[pyo3(get)]
+    pub t: Option<i64>,
+    #[pyo3(get)]
+    pub is_remote: bool,
```

Comment:
Is there any need for us to expose this in python

Reply:
no